src/format/mod.rs

```diff
@@ -378,6 +381,63 @@ const TOO_SHORT: ParseError = ParseError(ParseErrorKind::TooShort);
 const TOO_LONG: ParseError = ParseError(ParseErrorKind::TooLong);
 const BAD_FORMAT: ParseError = ParseError(ParseErrorKind::BadFormat);
 
+/// An error from a "format" function.
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+#[cfg(any(feature = "alloc", feature = "std", test))]
+pub struct FormatError(FormatErrorKind);
+
+/// The category of format error.
+#[cfg(any(feature = "alloc", feature = "std", test))]
+#[derive(Debug, Clone, PartialEq, Eq, Copy)]
+enum FormatErrorKind {
+    /// The locale is unknown or not available.
+    #[cfg(feature = "locales")]
+    UnknownLocale,
+
+    /// Format error.
+    Format,
```

Comment:
This should definitely include the source/cause format error so that we don't lose that.

Reply:
This has been reverted